Add documentation of the errors manager (#1395)
kuzzleio/kuzzle: doc/1/plugins/plugin-context/errors/kuzzleerror/index.md

## Properties

| Property   | Type                | Description                                                                                                           |
| ---------- | ------------------- | --------------------------------------------------------------------------------------------------------------------- |
| `message`  | <pre>string</pre>   | Error message                                                                                                         |

> | `stack`    | <pre>string[]</pre> | Error stack trace (not available in production mode)                                                                  |

Reviewer: ```suggestion
| `stack`    | <pre>string</pre> | Error stack trace (not available in production mode)                                                                  |
```

> | `status`   | <pre>integer</pre>  | Error status code, following the standard [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) 

Reviewer: ```suggestion
| `status`   | <pre>number</pre>  | Error status code, following the standard [HTTP status code](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) 
```

| `errorName`| <pre>string</pre>   | Error full name, composed by domain, subdomain and the name of the error |

> | `code`     | <pre>integer</pre>  | Error unique code |

Reviewer: ```suggestion
| `code`     | <pre>number</pre>  | Error unique code |
```

### errorName

> The `errorName` property is a concatenation of 3 levels of precision from where your error come from.

Reviewer 2: ```suggestion
The `errorName` property is a concatenation of 3 levels of precision; explaining where an error comes from.
```

> For an `errorName` `plugins.foobar-plugins.some_error`, you have `plugins` as domain, `foobar-plugins` as subdomain and `some_error` as error.

Reviewer 2: ```suggestion
For instance, an `errorName` set to `plugins.foobar-plugins.some_error` belongs to the `plugins` domain. It has been thrown by a plugin named `foobar-plugin`, and the error itself has been attributed the name of `some_error`.
```
